project/Build.scala

```diff
@@ -2155,51 +2163,66 @@ object Build {
     )
 
   lazy val `dist-mac-x86_64` = project.in(file("dist/mac-x86_64")).asDist(Bootstrapped)
+    .settings(packageName := (dist / packageName).value + "-x86_64-apple-darwin")
     .settings(
       republishBinDir := (dist / republishBinDir).value,
-      packArchiveName := (dist / packArchiveName).value + "-x86_64-apple-darwin",
       republishBinOverrides += (dist / baseDirectory).value / "bin-native-overrides",
       republishFetchCoursier := (dist / republishFetchCoursier).value,
       republishLaunchers +=
         ("scala-cli" -> s"gz+https://github.com/VirtusLab/scala-cli/releases/download/v$scalaCliLauncherVersion/scala-cli-x86_64-apple-darwin.gz")
     )
 
   lazy val `dist-mac-aarch64` = project.in(file("dist/mac-aarch64")).asDist(Bootstrapped)
+    .settings(packageName := (dist / packageName).value + "-aarch64-apple-darwin")
     .settings(
       republishBinDir := (dist / republishBinDir).value,
-      packArchiveName := (dist / packArchiveName).value + "-aarch64-apple-darwin",
       republishBinOverrides += (dist / baseDirectory).value / "bin-native-overrides",
       republishFetchCoursier := (dist / republishFetchCoursier).value,
       republishLaunchers +=
         ("scala-cli" -> s"gz+https://github.com/VirtusLab/scala-cli/releases/download/v$scalaCliLauncherVersion/scala-cli-aarch64-apple-darwin.gz")
     )
 
   lazy val `dist-win-x86_64` = project.in(file("dist/win-x86_64")).asDist(Bootstrapped)
+    .enablePlugins(WindowsPlugin) // TO GENERATE THE `.msi` installer
+    .settings(packageName := (dist / packageName).value + "-x86_64-pc-win32")
     .settings(
       republishBinDir := (dist / republishBinDir).value,
-      packArchiveName := (dist / packArchiveName).value + "-x86_64-pc-win32",
       republishBinOverrides += (dist / baseDirectory).value / "bin-native-overrides",
       republishFetchCoursier := (dist / republishFetchCoursier).value,
       republishExtraProps += ("cli_version" -> scalaCliLauncherVersion),
-      mappings += (republishRepo.value / "EXTRA_PROPERTIES" -> "EXTRA_PROPERTIES"),
```

Comment:
we need the `EXTRA_PROPERTIES` because it contains the scala-cli version to actually use on windows, unless we are not doing that `--cli-version` flag anymore? (in which case remove references in scripts)